contrib/CODINGSTYLE.md

```diff
@@ -0,0 +1,43 @@
+# Coding Style
+## Introduction
+This is the coding style guide for Modmesh.  
+* [Introduction](#introduction)
+* [Naming](#naming)
+* [Comments](#comments)
+* [Formatting](#formatting)
+* [Class](#class)
+* [Functions](#functions)
+
+### Naming
+1. The class names use Camel case, such as ```class R3DWidget;```. 
+2. The name of the variable / using-declarations uses snake case,  
+such as ```R3DWidget * viewer;``` and ```using size_type = std::size_t;```. 
+3. The QT-related function name uses Camel case,  
+whereas non-QT-related function name uses snake case.
+4. The name of the class member starts with ```m_```.
+5. The name of [macros](https://en.cppreference.com/w/cpp/preprocessor/replace) starts with ```DECL_MM_```.
```

Comment:
According to @tigercosmos and my suggestion,
here is the brief and rough conclusion.

However, it seems that the naming of the macros is inconsistent.
![image](https://github.com/solvcon/modmesh/assets/71295942/99a2985b-61f3-426f-8891-56fb196a9715)
Is there any naming rule about macros?

Reply:
interesting, @yungyuc  could you decide a prefix name and let's change all the other?